src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +29,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+
+        if (object.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) object;
+            return Objects.equals(width, rectangle.width)
+                    && Objects.equals(length, rectangle.length)
+                    && Objects.equals(color, rectangle.color);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 11;
+        result = 12 * width.hashCode();
+        result = 12 * length.hashCode();
+        result = 12 * color.hashCode();
```

Comment:
what if some of the fields would be null?

Reply:
will be panic))))